### DISCUSSION TITLE
[Question][Hw0-2] On Page Ranking Problem

### DISCUSSION THREAD

**Discussion by pingpingy1** [prosyslab-classroom/cs348-information-security#647] [Q&A]
Name: Heewon Lee

Hello. I have some questions regarding the specifications of the page ranking problem.
1. Can we assume that every page ID is distinct, i.e., no two pages have the same ID?
2. In case a certain page `u` has no outgoing links, should we assume that `B_u` includes `u` itself also? Or should we exclude the page?

Thank you.

**Comment by tanapthetimid** [comment]
I think judging from the test cases, every page id is distinct and B_u does not contain u itself.

**Comment by ICubE01** [comment]
1. ID means identity, so it's natural to assume that every page_id is unique.
2. The document says that "users can jump to arbitrary page on the web",  and also "N outgoing edges where N is the number of pages on the web". So a page should include itself when it has no outgoing link.

**Comment by pingpingy1** [comment]
Ah, thank you for clarifying!
